src/main/java/com/google/firebase/database/connection/NettyWebSocketClient.java

```diff
@@ -57,38 +60,51 @@
   private final ChannelHandler channelHandler;
   private final ExecutorService executorService;
   private final EventLoopGroup group;
+  private final boolean isSecure;
 
   private Channel channel;
 
   NettyWebSocketClient(
       URI uri, String userAgent, ThreadFactory threadFactory,
-      WebsocketConnection.WSClientEventHandler eventHandler) {
+      WebsocketConnection.WSClientEventHandler eventHandler, boolean isSecure) {
     this.uri = checkNotNull(uri, "uri must not be null");
     this.eventHandler = checkNotNull(eventHandler, "event handler must not be null");
     this.channelHandler = new WebSocketClientHandler(uri, userAgent, eventHandler);
     this.executorService = new FirebaseScheduledExecutor(threadFactory,
         "firebase-websocket-worker");
     this.group = new NioEventLoopGroup(1, this.executorService);
+    this.isSecure = isSecure;
   }
 
   @Override
   public void connect() {
     checkState(channel == null, "channel already initialized");
+    SslContext sslContext = null;
+    if (this.isSecure) {
+      TrustManagerFactory trustFactory = null;
+      try {
+        trustFactory = TrustManagerFactory.getInstance(
+            TrustManagerFactory.getDefaultAlgorithm());
+        trustFactory.init((KeyStore) null);
+        sslContext = SslContextBuilder.forClient()
+            .trustManager(trustFactory).build();
+      } catch (NoSuchAlgorithmException | KeyStoreException | SSLException e) {
+        eventHandler.onError(e.getCause());
+      }
+    }
     try {
-      TrustManagerFactory trustFactory = TrustManagerFactory.getInstance(
-          TrustManagerFactory.getDefaultAlgorithm());
-      trustFactory.init((KeyStore) null);
-      final SslContext sslContext = SslContextBuilder.forClient()
-          .trustManager(trustFactory).build();
       Bootstrap bootstrap = new Bootstrap();
       final int port = uri.getPort() != -1 ? uri.getPort() : DEFAULT_WSS_PORT;
+      final SslContext finalSslContext = sslContext;
```

Comment:
While super ugly, it is generally more comment to use a one element array for this:

```
  final SslContext[] sslContext = new SslContext[] {null};
  sslContext[0] = new ...
```